### PR TITLE
Add `persistent_workers` parameter to `TrainingArguments`

### LINKED ISSUE #27058: Add persistent_workers parameter to Dataloader
### Feature request

**persistent_workers** is a parameter that can be passed to Dataloader. It's set to **False** by default. What it does ([quote](https://discuss.pytorch.org/t/what-are-the-dis-advantages-of-persistent-workers/102110/2)):

> With this option to false, every time your code hits a line line for sample in dataloader:, it will create a brand new set of workers to do this loading and will kill them on exit.
> Meaning that if you have multiple dataloaders, the workers will be killed when you are done with one instantly.
> 
> If you make them persist, these workers will stay around (with their state) waiting for another call into that dataloader.
> 
> Setting this to True will improve performances when you call into the dataloader multiple times in a row (as creating the workers is expensive). But it also means that the dataloader will have some persistent state even when it is not used (which can use some RAM depending on your dataset).

Currently there is no option to pass **persistent_workers=True** to internal Dataloader without inheriting from **Trainer** class. Here is the code snippet from **trainer.py**:
  
```
def get_train_dataloader(self) -> DataLoader:
    """
    Returns the training [`~torch.utils.data.DataLoader`].

    Will use no sampler if `train_dataset` does not implement `__len__`, a random sampler (adapted to distributed
    training if necessary) otherwise.

    Subclass and override this method if you want to inject some custom behavior.
    """
    # ...
    # (some code)
    # ...
    return DataLoader(
        train_dataset,
        batch_size=self._train_batch_size,
        sampler=train_sampler,
        collate_fn=data_collator,
        drop_last=self.args.dataloader_drop_last,
        num_workers=self.args.dataloader_num_workers,
        pin_memory=self.args.dataloader_pin_memory,
        worker_init_fn=seed_worker,
    )
```

I suggest to add this parameter to **TrainingArguments** class. Set it to **False** by default to not break anyone's pipeline. 

### Motivation

Motivation is very simple: setting this parameter to **True** can speed up training process in some cases (but will require more RAM). 

### Your contribution

I can make the PR

### PR DESCRIPTION
**Pull request by Sorrow321**
# What does this PR do?

This PR gives the users the ability to pass `persistent_workers` parameters as `TrainingArguments` parameter to `Trainer`. Without this PR, you have to inherit from `Trainer` class to alter the behavior of `get_train_dataloader` function.

This parameters does the [following thing](https://pytorch.org/docs/stable/data.html): 

> If True, the data loader will not shut down the worker processes after a dataset has been consumed once. This allows to maintain the workers Dataset instances alive.

[Explanation](https://discuss.pytorch.org/t/what-are-the-dis-advantages-of-persistent-workers/102110/2?u=ifedorov):

> With this option to false, every time your code hits a line line for sample in dataloader:, it will create a brand new set of workers to do this loading and will kill them on exit.
> Meaning that if you have multiple dataloaders, the workers will be killed when you are done with one instantly.
> 
> If you make them persist, these workers will stay around (with their state) waiting for another call into that dataloader.
> 
> Setting this to True will improve performances when you call into the dataloader multiple times in a row (as creating the workers is expensive). But it also means that the dataloader will have some persistent state even when it is not used (which can use some RAM depending on your dataset).

Fixes [#27058](https://github.com/huggingface/transformers/issues/27058)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
It adds description of the parameter to the docs as well.
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
Yes
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
https://github.com/huggingface/transformers/issues/27058
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
Yes
- [ ] Did you write any new necessary tests?
No

## Who can review?

- trainer: @muellerzr @pacman100

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
LGTM, waiting for the approval let's see what @muellerzr thinks!

**Review by muellerzr — APPROVED** [review]
Thanks! LG2M as well

**Pull request merged**
